src/main/java/io/cryostat/net/web/http/api/v2/AbstractV2RequestHandler.java

```diff
@@ -119,46 +125,52 @@ protected Future<Boolean> validateRequestAuthorization(String authHeader) throws
 
     protected ConnectionDescriptor getConnectionDescriptorFromParams(RequestParameters params) {
         String targetId = params.getPathParams().get("targetId");
-        Credentials credentials = null;
-        if (params.getHeaders().contains(JMX_AUTHORIZATION_HEADER)) {
-            String proxyAuth = params.getHeaders().get(JMX_AUTHORIZATION_HEADER);
-            Matcher m = AUTH_HEADER_PATTERN.matcher(proxyAuth);
-            if (!m.find()) {
-                params.getHeaders().set(JMX_AUTHENTICATE_HEADER, "Basic");
-                throw new ApiException(427, "Invalid " + JMX_AUTHORIZATION_HEADER + " format");
-            } else {
-                String t = m.group("type");
-                if (!"basic".equals(t.toLowerCase())) {
+        try {
+            Credentials credentials = credentialsManager.getCredentialsByTargetId(targetId);
```

Comment:
If the credentials here were found by the `CredentialsManager`, is there a need to go through this whole if statement? If there was no such JMX_AUTH_HEADER, then would the ConnectionDescriptor be constructed anyways?

Reply:
This is supported intentionally to allow the client to override stored credentials by passing the `X-JMX-Authorization` header regardless. The ConnectionDescriptor will be constructed in any case, whether without credentials, with credentials from the header, or with stored credentials.

But, since checking for stored credentials is now a much more expensive operation than it used to be, I'll go through and refactor things so that we only perform that query if the header wasn't provided.